subsys/caf/modules/Kconfig.leds

```diff
@@ -15,6 +15,7 @@ if CAF_LEDS
 
 choice
 	prompt "Select LED driver"
+	default CAF_LEDS_GPIO if BOARD_NRF9160DK_NRF9160_NS
```

Comment:
I'm trying first if the CI goes through with this change.

Reply:
CI works fine with this change.
@pdunaj Can you check if this change would be fine in CAF LEDs module?

Reply:
I don't think this is a right way of handling it. Different applications may want to use different variant.
I think moving the configuration to Kconfig is a good idea but you should do it as was done in nrf_desktop cleanup.
Here for example you have set of BLE configs set depending on what is board capability: https://github.com/nrfconnect/sdk-nrf/blob/main/applications/nrf_desktop/Kconfig.ble#L40

asset tracker can place similar thing in https://github.com/nrfconnect/sdk-nrf/blob/main/applications/asset_tracker_v2/src/modules/Kconfig.led_module
e.g.,
config CAF_LEDS_GPIO
   default y if BOARD_NRF9160DK_NRF9160_NS

Reply:
I agree with @pdunaj. I think we should not introduce board nor application specific dependencies to the CAF LEDs module.

Reply:
CAF_LEDS_GPIO is a choice symbol for which you cannot define a default value. I haven't found a way to do a redefinition for a choice symbol but I get the following errors:

`warning: default on the choice symbol CAF_LEDS_GPIO (defined at /home/tor1/work/ncs/nrf/applications/asset_tracker_v2/src/modules/Kconfig.led_module:32, /home/tor1/work/ncs/nrf/subsys/caf/modules/Kconfig.leds:24) will have no effect, as defaults do not affect choice symbols
`

`warning: the choice symbol CAF_LEDS_GPIO (defined at /home/tor1/work/ncs/nrf/applications/asset_tracker_v2/src/modules/Kconfig.led_module:32, /home/tor1/work/ncs/nrf/subsys/caf/modules/Kconfig.leds:24) is defined with a prompt outside the choice
`

If I try to redefine the entire choice, I also get the following error:

`
warning: the default selection CAF_LEDS_GPIO (defined at /home/tor1/work/ncs/nrf/applications/asset_tracker_v2/src/modules/Kconfig.led_module:32, /home/tor1/work/ncs/nrf/subsys/caf/modules/Kconfig.leds:24) of <choice> (defined at /home/tor1/work/ncs/nrf/applications/asset_tracker_v2/src/modules/Kconfig.led_module:20) is not contained in the choice
`

Reply:
In order to redefine the choice symbol it will need a name.
Something like this:

```
choice **CAF_LEDS_OUTPUT**
	prompt "Select LED driver"
	default CAF_LEDS_PWM

config CAF_LEDS_PWM
	bool "PWM"
	depends on LED_PWM

config CAF_LEDS_GPIO
	bool "GPIO"
	depends on LED_GPIO

endchoice
```

Then you should be able to do:
```
choice CAF_LEDS_OUTPUT
	default CAF_LEDS_PWM if BOARD_THINGY91_NRF9160_NS
	default CAF_LEDS_GPIO
endchoice
```